RSA.xs

```diff
@@ -55,16 +86,24 @@ void croakSsl(char* p_file, int p_line)
 
 char _is_private(rsaData* p_rsa)
 {
-    const BIGNUM *d;
 #if OLD_CRUFTY_SSL_VERSION
+    const BIGNUM* d;
     d = p_rsa->rsa->d;
 #else
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+    BIGNUM* d = NULL;
+    EVP_PKEY_get_bn_param(p_rsa->rsa, OSSL_PKEY_PARAM_RSA_D, &d);
+#else
+    const BIGNUM* d;
     RSA_get0_key(p_rsa->rsa, NULL, NULL, &d);
 #endif
-    return(d != NULL);
+#endif
+    char ret = (d != NULL);
+    BN_clear_free(d);
```

Comment:
This needs an `#if OPENSSL_VERSION_NUMBER >= 0x30000000L` guard. For older versions that's an internal pointer that needs to remain intact.

Reply:
Thanks for the help on this @botovq